esiclient/v1/node_network.py

```diff
@@ -264,21 +264,48 @@ def take_action(self, parsed_args):
         ironic_client = self.app.client_manager.baremetal
         neutron_client = self.app.client_manager.network
 
-        node = None
-        port = None
+        ports = list((port for port in ironic_client.port.list(detail=True)))
```

Comment:
`ironic_client.port.list` should take a node parameter that limits the results to the baremetal ports associated with the node; you can see an example at https://github.com/CCI-MOC/python-esiclient/blob/master/esiclient/v1/node_network.py#L52. Doing so should remove the need to manually filter the port list below.

Also, if the user passes in a port, there's no need to do this, so you probably want to check that first - something like:

```
if parsed_args.port:
   port_uuid = parsed_args.port
else:
   <find the port_uuid by listing the baremetal ports>
```